File: workspace/src/my_nav_pkg/src/my_nav_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

// using std::placeholders::_1;

class MyNavNodeCpp : public rclcpp::Node
{
public:
  MyNavNodeCpp() : Node("my_nav_node_cpp"), count_(0)
  {
    publisher_ = this->create_publisher<std_msgs::msg::String>("chatter_cpp", 10);
    timer_ = this->create_wall_timer(
      std::chrono::milliseconds(500),
      std::bind(&MyNavNodeCpp::timer_callback, this));
  }

private:
  void timer_callback()
  {
    auto msg = std_msgs::msg::String();
    msg.data = "Hello from C++ node " + std::to_string(count_++);
    publisher_->publish(msg);
    RCLCPP_INFO(this->get_logger(), "%s", msg.data.c_str());
  }

  rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
  int count_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<MyNavNodeCpp>());
  rclcpp::shutdown();
  return 0;
}

File: workspace/src/my_nav_pkg/my_nav_pkg/my_nav_node.py
#! /usr/bin/env python3

import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class MyNavNode(Node):
    def __init__(self):
        super().__init__('my_nav_node')
        self.publisher_ = self.create_publisher(String, 'chatter', 10)
        self.timer = self.create_timer(0.5, self.timer_callback)  # 2Hz
        self.count = 0

    def timer_callback(self):
        msg = String()
        msg.data = f"Hello from Python node {self.count}"
        self.publisher_.publish(msg)
        self.get_logger().info(msg.data)
        self.count += 1

def main(args=None):
    rclpy.init(args=args)
    node = MyNavNode()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()
